### PR TITLE
Support multi-dim reduce op

### PR DESCRIPTION
**Pull request by barrh**
Prior to the patch, networks that apply reduce operation on
multiple dimensions have failed on ONNX JIT pass during
graph optimization.
e.g. torch.mean(tensor, [dim1, dim2])

This patch adds support for iterable dim argument.

Resolves issue #20942

### CODE REVIEW

File: torch/onnx/symbolic.py [hunk 1]
```diff
@@ -80,7 +80,13 @@ def _parse_arg(value, desc):
         elif desc == 't':
             return tval
         elif desc == 'is':
-            return [int(v) for v in tval]
+            try:
+                return [int(v) for v in tval]
+            except TypeError:
+                if tval.dim():
```
Comment (ifedan) [inline torch/onnx/symbolic.py]: What is the purpose of calling dim() and raise the exceptions? Are you trying to figured out that tval is Tensor?

Comment (barrh) [reply torch/onnx/symbolic.py]: Actually, I'm kinda assuming it's a tensor at that point. Trying to iterate over zero-dimension tensor throws a TypeError, so if the exception was raised due to that edge-case, I silence it.

**Comment by barrh** [comment]
@yinghai would you mind approving this change?

File: torch/onnx/symbolic.py [hunk 2]
```diff
@@ -80,7 +80,13 @@ def _parse_arg(value, desc):
...
+                    raise
+                # scalar tensors are not iterable
+                return [int(tval)]
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: Hi @barrh, this is not the right way to handle different type of inputs. You can use instance() function or _is_value + lengths.type().isSubtypeOf(torch._C.TensorType.get()). There are couples of examples in symbolic.py. Also, please add an end to end test case to test_pytorch_onnx_caffe2.py. We don't encourage adding more test cases to test_operators.py if unnecessary.

File: torch/onnx/symbolic.py [hunk 3]
```diff
@@ -1195,7 +1202,7 @@ def feature_dropout(g, input, p, train):
 feature_alpha_dropout_ = feature_alpha_dropout
 
 
-@parse_args('v', 't', 'i', 'i')
+@parse_args('v', 't', 'is', 'i')
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: Why change this?

**Review by houseroad — CHANGES_REQUESTED** [review]
Please address the inline comments.

File: torch/onnx/symbolic.py (hunk 3 above)
Comment (barrh) [reply torch/onnx/symbolic.py]: Similarly to mean, [torch.norm](https://pytorch.org/docs/stable/torch.html#torch.norm) accepts tuple types.
Before this change, `torch.norm(d, dim=(1,2))` would have failed.

Comment (houseroad) [reply torch/onnx/symbolic.py]: if so, can we also add test cases?

Comment (barrh) [reply torch/onnx/symbolic.py]: Similarly to mean, [torch.norm](https://pytorch.org/docs/stable/torch.html#torch.norm) accepts tuple types.
Before this change, torch.norm(d, dim=(1,2)) would have failed.

File: torch/onnx/symbolic.py (hunk 2 above)
Comment (barrh) [reply torch/onnx/symbolic.py]: @houseroad  I didn't understand your comment regarding the handling.
To allow this handling scale beyond 1 element, I iterate the tensor, but catch exception for scalar tensors (e.g. `torch.tensor(1)`. The TypeError is raised due to the traversing on non-iterable, not the element's type.

File: torch/onnx/symbolic.py (hunk 3 above)
Comment (barrh) [reply torch/onnx/symbolic.py]: added two tests.

**Comment by barrh** [comment]
@houseroad Do you have any idea how can I resolve the CI issues?
Not sure why I get the segfault, but it's pretty consistent, so it has to be introduced by my patch.
